### PR TITLE
Bump rust-openssl, add support for OpenSSL 1.1.0 [WIP]

### PR DESCRIPTION
**Pull request by frewsxcv**
Fixes https://github.com/servo/servo/issues/14203.

TODO:

* Stop using a git dep and use this published version: https://github.com/nox/hyper_serde/pull/2
 * Or maybe remove hyper_serde

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14484)

### CODE REVIEW

File: components/net/connector.rs [hunk 1]
```diff
@@ -28,33 +29,18 @@ const DEFAULT_CIPHERS: &'static str = concat!(
...
-        SslStream::connect(ssl, stream).map_err(From::from)
+    let ca_file = &resources_dir_path()
+        .expect("Need certificate file to make network requests")
+        .join("certs");
+    let mut ssl_connector_builder = SslConnectorBuilder::new(SslMethod::tls()).unwrap();
+    {
+        let ssl_context_builder = ssl_connector_builder.builder_mut();
+        ssl_context_builder.set_ca_file(ca_file).expect("could not set CA file");
+        ssl_context_builder.set_cipher_list(DEFAULT_CIPHERS).expect("could not set ciphers");
+        ssl_context_builder.set_options(SSL_OP_NO_SSLV2 | SSL_OP_NO_SSLV3 | SSL_OP_NO_COMPRESSION);
```
Comment (sfackler) [inline components/net/connector.rs]: These are all set already by the connector fyi: https://github.com/sfackler/rust-openssl/blob/master/openssl/src/ssl/connector.rs#L29

**Comment by avadacatavra** [comment]
i'm working on rebasing and updating this

**Comment by fkautz** [comment]
@avadacatavra a few updates which may help you

* hyper 0.10 was released, removes deps on cookie and openssl packages
* webdriver still depends on openssl 0.7 and cookie 0.2
* cookie 0.2 depends on openssl 0.7, but cookie 0.4 and above already have openssl 0.9 support

**Comment by jdm** [comment]
Not much point in leaving this open. @avadacatavra will open a new PR, presumably.

**Pull request closed without merging**